### ISSUE TITLE
webpack 5 can't set output path to `/`

### ISSUE THREAD

**Issue by sodatea** [webpack/webpack#10544]
# Bug report

**What is the current behavior?**

When the `output.path` is set to `/`, webpack 5 will fail with an `EISDIR` error. It works in webpack 4.
`/` is commonly used when targeting an in-memory filesystem such as https://github.com/streamich/memfs; we used it quite often in tests.

**If the current behavior is a bug, please provide the steps to reproduce.**

```sh
git clone https://github.com/sodatea/webpack-rootdir-bug.git
cd webpack-rootdir-bug
yarn && yarn build
```

**What is the expected behavior?**

Should successfully build.

**Other relevant information:**
webpack version: v5.0.0-beta.14
Node.js version: v12.16.1
Operating System: macOS Catalina 10.15.3
Additional tools: None

**Comment by evilebottnawi** [comment]
Yep, `memory-fs` for dev server also has that limitation, what is use case?

**Comment by sodatea** [comment]
See the linked pull request.

`memory-fs` throws `EEXIST`, which webpack's `fs.mkdirp` implementation could recognize.
But BSD systems and `memfs` throws `EISDIR`. The refactored `fs.mkdirp` in webpack 5 couldn't handle that error and fails.

**Comment by evilebottnawi** [comment]
bump

**Comment by evilebottnawi** [comment]
bump

**Comment by alexander-akait** [comment]
Still valid
